Find a functionally equivalent code:android.app.Activity.shouldUpRecreateTask:COMMENT
Method Modifier: public      
Comment:/**
 * Returns true if the app should recreate the task when navigating 'up' from this activity
 * by using targetIntent.
 *
 * <p>If this method returns false the app can trivially call
 * {@link #navigateUpTo(Intent)} using the same parameters to correctly perform
 * up navigation. If this method returns false, the app should synthesize a new task stack
 * by using {@link TaskStackBuilder} or another similar mechanism to perform up navigation.</p>
 *
 * @param targetIntent An intent representing the target destination for up navigation
 * @return true if navigating up should recreate a new task stack, false if the same task
 * should be used for the destination
 */

@@ -9,15 +9,15 @@
         if (info.taskAffinity == null) {
             return false;
         }
-        return !ActivityManagerNative.getDefault().targetTaskAffinityMatchesActivity(mToken, info.taskAffinity);
+        return ActivityManagerNative.getDefault().shouldUpRecreateTask(mToken, info.taskAffinity);
     } catch (RemoteException e) {
         return false;
     } catch (NameNotFoundException e) {

